cmd/pathlist.go

```diff
@@ -0,0 +1,83 @@
+package cmd
+
+import 	(
+	"fmt"
+	"os"
+	"regexp"
+	"bufio"
+	"io"
+	"sort"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	regex = "data/[A-Za-z0-9_./]+\\.[a-zA-Z0-9]+"
+)
+
+var pattern regexp.Regexp
+
+func init() {
+	pathlistCmd.AddCommand(pathlistScanCmd())
+	pattern = *regexp.MustCompile(regex)
+}
+
+var pathlistCmd = &cobra.Command{
+	Use:   "pathlist",
+	Short: "Work with pathlist files",
+}
+
+func pathlistScanCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use: "scan",
+		Short: "Scan a core file for interesting paths",
+		Args: cobra.ExactArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			verbose, _ := cmd.PersistentFlags().GetBool("verbose")
+			reader, openErr := os.Open(args[0])
+			r := bufio.NewReader(reader)
+			if openErr != nil {
+				fmt.Fprintln(os.Stderr, openErr)
+				os.Exit(1)
+			}
+			matches := make(map[string]struct{})
+			offset := int64(0)
+			for {  // Inspired from https://go.dev/play/p/aPrAW7XGHi
+
+				m := pattern.FindReaderIndex(r)
+				if m == nil {
+					break
+				}
+
+				offset += int64(m[0])  // Add the offset.
+				reader.Seek(offset, 0)  // Seek to the match.
+				r.Reset(reader)  // Flush buffer so buffered reader is looking where we want it to.
+
+				b := make([]byte, m[1]-m[0])
+				io.ReadFull(r, b)
+				s := string(b[:])
+				if verbose {
+					fmt.Fprintf(os.Stderr, "offset=%v s=%v\n", offset, s)
+				}
+
+				matches[s] = struct{}{}  // Set membership
+				offset += int64(m[1] - m[0])  // Keep record of where we think the reader is looking next.
+			}
+			if verbose {
+				fmt.Fprintf(os.Stderr, "\nFound %v matches.\n", len(matches))
+			}
+			ary := []string{}
+			for path, _ := range matches {
+				ary = append(ary, path)
+			}
+
+			sort.Strings(ary)
```

Comment:
What's the point of sorting the list prior to printing? If we _don't_ sort, then we can just print items as they're detected, instead of having to build a list in-memory and then print it all at the end. (And the caller can always just pipe to `sort` if they want)

Reply:
This is to encourage a standard ordering for the file listings.  People in the modding community have varying stanadrds, so it's good to enforce some at the program level.

Reply:
There's a windows user base too, and sorting text in powershell is cock n ball torture - it'll add CRLF line endings if you're not careful... so best to do it for them

Reply:
Fair enough--maybe add a flag to disable sorting, then? Or maybe not sorting the default and add a flag to sort. Just so we have _some_ way to control it

Reply:
could we make an issue to add more control over the output?

Reply:
Sounds good to me!